Server/src/BaseQuasarServer.cpp

```diff
@@ -198,6 +204,73 @@ std::string BaseQuasarServer::getApplicationPath() const
     return std::string(serverSettingsPath);
 }
 
+std::string BaseQuasarServer::getProcessEnvironmentVariables() const
+{
+    std::ostringstream result;
+#ifdef __linux__
+    extern char **environ;
+    char *s = *environ;
+    for (int i = 0; s; ++i)
+    {
+        result << s << std::endl;
+        s = *(environ + i);
+    }
+#elif _WIN32
+    auto envBlockDeleterFn = [](LPTCH p) { FreeEnvironmentStrings(p); };
+    auto envBlock = std::unique_ptr<TCHAR, decltype(envBlockDeleterFn)>{ GetEnvironmentStrings(), envBlockDeleterFn };
+    // envBlock format: "keyA=valA\0keyB=valB\0...keyXXX=valXXX\0\0". Double \0\0 denotes end of block
+    for (TCHAR* pos = envBlock.get(); pos != nullptr && *pos != TCHAR('\0'); )
+    {
+        const std::string envVar(pos); // capture from pos up to '\0'
+        result << envVar << std::endl;
+        pos += envVar.length() + 1; // +1 ? wind pos past '\0' at end of current key/val pair
+    }
+#endif
+    return result.str();
+}
+
+/**
+* Returns string in format: logged in user (effective process owner).
+* Examples, where a user 'quasar' is logged in...
+* == posix ==
+* 'quasar' runs process regular - returns 'quasar(quasar)'
+* 'quasar' runs process with sudo - returns 'quasar(root)'
+* == windows ==
+* 'quasar' runs process regular - returns 'quasar(normal)'
+* 'quasar' runs process with admin rights (or admin shell) 'quasar(elevated)'
+*/
+std::string BaseQuasarServer::getProcessOwner() const
+{
+    std::ostringstream result;
+#ifdef __linux__
+    const auto userID = getlogin();
+    result << (userID != nullptr ? userID : "unknown");
+
+    std::string effectiveUserID = "unknown";
+    auto pwuid = getpwuid(getuid());
+    if (pwuid != nullptr && pwuid->pw_name != nullptr)
+    {
```

Comment:
Asking for improvement: while information coming from getuid() and geteuid() is always available (reference: "man getuid") while the numeric user ID might be not mappable to its username. It actually is a common situation in distributed systems. Please improve it such that the numeric UID + numeric EUID is always printed, while respective usernames (for both UID or EUID) are printed *in addition*, if they can be obtained.

Reply:
Good idea, will do

Reply:
Ah, irritatingly not so straightforward as I thought, see [this comment](https://github.com/quasar-team/quasar/pull/232#discussion_r522852521) regarding getuid and geteuid actually return. 

I mean `getlogin()`  (returns string) does not refer to same user as returned by `getuid()`, as was my (naive) hope.

And just for general hilarity, stay well clear of method `cuserid` apparently, man cuserid... 
"Nobody knows precisely what cuserid() does; avoid it in portable programs.  Or avoid it altogether"

Reply:
Update: added (numerical) result of geteuid to the output string. So now looks like

* `Process owner: bfarnham(56850:bfarnham)` (run normal privs)
* `Process owner: bfarnham(0:root)` (run as sudo)

Reply:
Thanks!